pkg/trace/api/ratelimiter.go

```diff
@@ -102,7 +102,7 @@ func (ps *rateLimiter) RealRate() float64 {
 func (ps *rateLimiter) realRateLocked() float64 {
 	if ps.stats.RecentTracesSeen <= 0 {
 		// avoid division by zero
-		return ps.stats.TargetRate
+		return 1
```

Comment:
I think you also need to work on this part of the code:

https://github.com/DataDog/datadog-agent/blob/eab4634fe177e31dfb05b3352e797a70bd97c9c0/pkg/trace/api/ratelimiter.go#L130

Because now that it returns a high (1) value, agree it's more consistent, but the "first trace" will not show up, as I understand it. Which is sad because when you receive no trace, the 1st one that shows -> you want to keep it for sure.

Reply:
Thanks for catching this. You're right. This does need more work. I'll have to think of a good solution. I'm not sure if this one is it.